docs/TrafficSplitting.md

```diff
@@ -0,0 +1,166 @@
+2. `--tag`: Second, all the revisions are tagged as specified in the traffic block
+3. `--traffic`: Third, all the referenced revisions (or tags) are assigned given portion of traffic split
+
+*Note*:
+- Use comma separated value pairs or separate value pairs:
+  For example: `--tag echo-v1=v1,echo-v2=v2` is same as `--tag echo-v1=v1 --tag echo-v2=v2`
+  similarly, `--traffic v1=50,v2=50` is same as `--traffic v1=50 --traffic v2=50`
+  also, `--untag v1,v2` is same `--untag v1 --untag v2`
+- `--untag` flag's action takes first precedence. If `--untag` and `--tag` flags specified together for revisions, all the mentioned tags will be unassigned from revisions first, followed by tagging them.
+- Updating a tag for Revision requires untagging its existing tag and assigning a new tag.
+  For example, to update tag from 'testing' to 'staging' for Revision 'echo-v1': `--untag testing --tag staging`
+- If a tag is assigned to a Revision and user tries to use same tag for another Revision, an error is raised.
+- If a tag is assigned to a Revision and user tries to tag the same Revision with a new tag again, a new target is created with 0% traffic and new tag.
+- To assign multiple tags to same Revision, do `--tag sameRevisionName=tagName1 --tag sameRevisionName=tagName2`.
+- Repetition of `@latest` identifier for `--traffic` and `--tag` flags is not allowed.
+- Repetition of revisionName or tagName for `--traffic` flag is not allowed;
+  - If you want to have same revision present in multiple targets, use different tags to uniquely identify targets,
+    and use unique tag names to assign traffic portions.
+    For example: `kn service update svc --tag echo-v1=v1,echo-v2=v2, --traffic v1=50,v2=50`
+- If one traffic block target has revisionName which matches tagName of another target, tagName is looked up first for assigning traffic.
+  For example, `--traffic targetRef:percent` will assign mentioned `percent` to target where tag is `targetRef`.
+  **Tags are unique** in traffic block, **RevisionNames are not**, use tag reference in such scenarios.
+
+
+## Summary
+Following table shows quick summary for traffic splitting flags, their value formats, operation to perform and *Repetition* column denotes if repeating the particular value of flag is allowed in a `kn service update` command.
+
+
+Flag | Value(s) | Operation | Repetition
+:--- | :---: | :--- | :---:
+`--traffic` | `RevisionName=Percent` | Give `Percent` traffic to `RevisionName` | :heavy_check_mark:
+`--traffic` | `Tag=Percent` | Give `Percent` traffic to the Revision having `Tag` | :heavy_check_mark:
+`--traffic` | `@latest=Percent` | Give `Percent` traffic to the latest ready Revision | :heavy_multiplication_x:
+`--tag` | `RevisionName=Tag` | Give `Tag` to `RevisionName` | :heavy_check_mark:
+`--tag` | `@latest=Tag` | Give `Tag` to the latest ready Revision | :heavy_multiplication_x:
+`--untag` | `Tag` | Remove `Tag` from Revision | :heavy_check_mark:
+
+
+## Examples
```

Comment:
Also great. But perhaps better above

Reply:
I added an example per sub-section of flag description, the examples at the bottom refers to specific use cases a user might have and respective commands.